packages/analytics/amplify_analytics_pinpoint_dart/test/queued_item_store_test.dart

```diff
@@ -0,0 +1,299 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/dart_queued_item_store.dart';
+
+import 'package:test/test.dart';
+
+void main() {
+  late DartQueuedItemStore db;
+
+  group('DartQueuedItemStore ', () {
+    setUpAll(() {
+      db = DartQueuedItemStore('test');
+    });
+
+    setUp(() async {
+      await db.clear();
+    });
+
+    test('writes values to storage', () async {
+      const values = ['0', '1', '2', '3', '4', '5'];
+      for (final value in values) {
+        await db.add(value);
+      }
+
+      final readItems = await db.get(6);
```

Comment:
*nit:* It isn't immediately obvious what the significance of 6 is.
```suggestion
      final readItems = await db.get(values.length);
```